### ISSUE TITLE
Autocomplete for a class instance in a numpy array

### ISSUE THREAD

**Issue by sigbjorngrini** [nteract/hydrogen#1035]
**Description:**

Autocomplete is not activated equally for an instance of a class when it is an element in a numpy array compared to an element in a list.

**Steps to Reproduce:**

```Python
#%% 

class Car(object):
    def __init__(self, name):
        self.name = name
        self.backward_name = name[::-1]

#%%

car = Car('test')

#%%
#### Press ctrl-space after car.
car.

#%%
my_cars = [Car('test'), Car('test2')]
c = my_cars[0]
    
#%%
#### Press ctrl-space after c.
c.

#%%

cars = np.array([Car('test'), Car('test2')])
d = cars[0]

#%%
#### Press ctrl-space after d.
d.

```

Typing ctrl-space at car. and c. gives you possible options, while for at d. you get .self as possibility. If you start to write d.na the option will come up. Sorry if this is not a hydrogen-related issue. Thank you for the work.

**Versions:**

Windows 10
Atom: 1.21.0
Electron: 1.6.9

**Comment by lgeiger** [comment]
Hydrogen only offers autocompletion for words longer than 3 characters.

Should we make Hydrogen respect the global autocomplete setting? It should be accessible using 
```js
atom.config.get('autocomplete-plus.minimumWordLength')
```

**Comment by sigbjorngrini** [comment]
Thank you @lgeiger for the quick reply. I wasn't aware of that.  The point I wanted to make also is that they give different autocomplete options, but I realized now that this is an autocomplete-plus problem. This is what I get now with longer than 3 characters:

![list_car](https://user-images.githubusercontent.com/10991790/31352516-54d82d9e-ad2f-11e7-9e1f-6fc1dae40fd8.PNG)

![array_car](https://user-images.githubusercontent.com/10991790/31352519-58676e2a-ad2f-11e7-8868-c5b94594a5be.PNG)

**Comment by lgeiger** [comment]
The completions you are seeing in the list case aren't from Hydrogen. They're probably coming from `autocomplete-python` or similar.
It looks like the completion results of Hydrogen are hidden underneath.

**Comment by sigbjorngrini** [comment]
Yes they are, so it is ```autocomplete-python``` that doesn't show for the numpy case. Thank you. Anyhow it would be nice to have autocompletion for shorter that 3 characters for my use, but I understand that it may not be preferable for others.

**Comment by lgeiger** [comment]
>  Anyhow it would be nice to have autocompletion for shorter that 3 characters for my use, but I understand that it may not be preferable for others.

@sigbjorngrini Would you be interested in sending a PR to fix this? We're happy to provide help with it.

`autocomplete-plus` has a config option for this which is accessible via `atom.config.get('autocomplete-plus.minimumWordLength')`. We do the check for the character length here:
https://github.com/nteract/hydrogen/blob/2498e5ed21e62219d59a66511207eb48d8fa5b6b/lib/autocomplete-provider.js#L110

**Comment by sigbjorngrini** [comment]
@lgeiger Ok, I'll try fix this and send a PR.

**Comment by lgeiger** [comment]
@sigbjorngrini That's great!

Feel free to join our [Slack](https://slackin-nteract.now.sh/) if you have any questions.

**Comment by sigbjorngrini** [comment]
Solved with #1054
